pkg/resource/deploy/plan.go

```diff
@@ -60,6 +64,36 @@ type PlanSummary interface {
 	Sames() map[resource.URN]bool
 }
 
+// InvalidResourceError is returned by deploy.NewPlan when the engine observes
+// that one or more resources are in an invalid state. The engine populates `InvalidResources`
+// with the URNs of every resource that was invalid.
+type InvalidResourceError struct {
+	InvalidResources []*resource.State
+}
+
+func (ire InvalidResourceError) Error() string {
+	var buf bytes.Buffer
+	writer := bufio.NewWriter(&buf)
+	fmt.Fprintf(writer,
+		"error: the current deployment has %d resource(s) whose statuses are unknown:\n", len(ire.InvalidResources))
+
+	for _, res := range ire.InvalidResources {
+		fmt.Fprintf(writer, "  * %s, interrupted while %s\n", res.URN, res.Status)
+	}
+
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "These resources have unknown statuses because the Pulumi CLI was interrupted before it\n")
```

Comment:
Unfortunate that we have to reference the Pulumi CLI here. Any chance we can move that up-stack?

As far as wordsmithing goes:
"These resources are in an unknown state because the Pulumi CLI was interrupted while waiting for changes to these resources to complete. You should confirm whether or not the operations listed above completed successfully by checking the state of the appropriate provider (e.g. for an AWS resource, examining the resource's state in the AWS console).

Once you have checked the state of the resources, you can repair your stack..."